code-challenges.test.js

```diff
@@ -25,14 +25,29 @@ var people = [
 ]
 // Expected output: ["Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is a president of the galaxy.", "Arthus Dent is a radio employee."]
 
+// the function greeting is defined in the describe portion.
+describe("greeting", () => {
 
+  // describe in plain words what we want the function to execute.
+  test("return a string with a greeting sentence capitalizing the value of the name key", () => {
 
-
+    //the exact expected outcome is in the .toEqual argument.
+    expect(greeting(people)).toEqual("Ford Prefect is a hitchhiker.", "Zaphod Beeblebrox is a president of the galaxy.", "Arthus Dent is a radio employee")
+    
+  })
+})
 
 // b) Create the function that makes the test pass.
-
-
-
+// create a function that takes an array of objects and capitalizes the value of the name
+// map over object people to uppercase the name value 
+const greeting = (person) => {
+  obj.map(function (item)) {
+    // I found the Object.assign method that could be used here.
+    let newObj = Object.assign(item, {name: item.name.toUperCase())}
+    // string interpolation to get the values passed in
+      return `${item} is a ${occupation}`
+  }
+}
```

Comment:
I couldn't get this function to run as is. Were you able too? I can see the logic you're following and I like the path you're on. Returning a .map() will give us a new array and I like the usage of string interpolation in it.